go/src/koding/klient/machine/index/index.go

```diff
@@ -365,5 +424,43 @@ func (idx *Index) UnmarshalJSON(data []byte) error {
 	}
 	defer r.Close()
 
-	return json.NewDecoder(r).Decode(&idx.root)
+	if err = json.NewDecoder(r).Decode(&idx.root); err != nil {
+		return err
+	}
+
+	// BUG(rjeczalik): Something overwrites the root entry
+	// with a zero value elsewhere. Fix me.
+	idx.root.Entry = newEntry()
```

Comment:
it is "fixed"(actually it wasn't a bug) but not deployed on my testing machine. Lets not remove this until all klients have updated `newEntry` logic.